src/control/destination/background/mod.rs

```diff
@@ -318,12 +320,13 @@ where
 
 impl NewQuery {
 
-    fn new(namespaces: Namespaces, suffixes: Vec<dns::Suffix>, concurrency_limit: usize) -> Self {
+    fn new(namespaces: Namespaces, suffixes: Vec<dns::Suffix>, concurrency_limit: usize, proxy_id: String) -> Self {
         Self {
             namespaces,
             suffixes,
             concurrency_limit,
             active_query_handle: Arc::new(()),
+            proxy_id,
```

Comment:
Since this is cloned into every request, I wonder if it ought to be reference-counted?